llvm/lib/Target/SPIRV/SPIRVOperands.cpp

```diff
@@ -0,0 +1,178 @@
+             Category == ::OperandCategory::LoopControlOperand ||
+             Category == ::OperandCategory::FunctionControlOperand ||
+             Category == ::OperandCategory::MemorySemanticsOperand ||
+             Category == ::OperandCategory::MemoryOperandOperand ||
+             Category == ::OperandCategory::KernelProfilingInfoOperand) {
+    // Value that encodes many enum values (one bit per enum value)
+    std::string name;
+    std::string separator;
+
+    const SymbolicOperand *enumValueInCategory =
+        lookupSymbolicOperandByCategory(Category);
+
+    while (enumValueInCategory && enumValueInCategory->Category == Category) {
+      if ((enumValueInCategory->Value != 0) &&
+          (Value & enumValueInCategory->Value)) {
+        name += separator + enumValueInCategory->Mnemonic.str();
+        separator = "|";
+      }
+
+      ++enumValueInCategory;
+    }
+
+    return name;
+  }
+
+  return "UNKNOWN";
+}
+
+uint32_t
+getSymbolicOperandMinVersion(::OperandCategory::OperandCategory Category,
+                             uint32_t Value) {
+  const SymbolicOperand *lookup =
+      lookupSymbolicOperandByCategoryAndValue(Category, Value);
+
+  if (lookup)
+    return lookup->minVersion;
+
+  return 0;
+}
+
+uint32_t
+getSymbolicOperandMaxVersion(::OperandCategory::OperandCategory Category,
+                             uint32_t Value) {
+  const SymbolicOperand *lookup =
+      lookupSymbolicOperandByCategoryAndValue(Category, Value);
+
+  if (lookup)
+    return lookup->maxVersion;
+
+  return 0;
+}
+
+std::vector<::Capability::Capability>
+getSymbolicOperandCapabilities(::OperandCategory::OperandCategory Category,
+                               uint32_t Value) {
+  const CapabilityEntry *capability =
+      lookupCapabilityByCategoryAndValue(Category, Value);
+
+  std::vector<::Capability::Capability> capabilities;
+  while (capability && capability->Category == Category &&
+         capability->Value == Value) {
+    capabilities.push_back(
+        static_cast<::Capability::Capability>(capability->ReqCapability));
+    ++capability;
+  }
+
+  return capabilities;
+}
+
+std::vector<::Extension::Extension>
+getSymbolicOperandExtensions(::OperandCategory::OperandCategory Category,
+                             uint32_t Value) {
+  const ExtensionEntry *extension =
+      lookupExtensionByCategoryAndValue(Category, Value);
+
+  std::vector<::Extension::Extension> extensions;
+  while (extension && extension->Category == Category &&
+         extension->Value == Value) {
+    extensions.push_back(
+        static_cast<::Extension::Extension>(extension->ReqExtension));
+    ++extension;
+  }
+
+  return extensions;
+}
+
+std::string getLinkStringForBuiltIn(::BuiltIn::BuiltIn BuiltInValue) {
+  const SymbolicOperand *lookup = lookupSymbolicOperandByCategoryAndValue(
+      ::OperandCategory::BuiltInOperand, BuiltInValue);
+
+  if (lookup)
+    return "__spirv_BuiltIn" + lookup->Mnemonic.str();
```

Comment:
I'd say we need to have this string constant unique somewhere to be used by both OCLBIFs and here.

And I'm strongly against this '15' constant below, won't hurt using this constant string length instead